firefox_puppeteer/ui/pageinfo/deck.py

```diff
@@ -0,0 +1,202 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By, Wait
+
+from ...base import UIBaseLib
+
+
+class Deck(UIBaseLib):
+
+    def _create_panel_for_id(self, panel_id):
+        """Creates an instance of :class:`Panel` for the specified panel id.
+
+        :param panel_id: The ID of the panel to create an instance of.
+
+        :returns: :class:`Panel` instance
+        """
+        mapping = {'feedPanel': FeedPanel,
+                   'generalPanel': GeneralPanel,
+                   'mediaPanel': MediaPanel,
+                   'permPanel': PermissionsPanel,
+                   'securityPanel': SecurityPanel
+                   }
+
+        panel = self.element.find_element(By.ID, panel_id)
+        return mapping.get(panel_id, Panel)(lambda: self.marionette, self.window, panel)
+
+    # Properties for visual elements of the deck #
+
+    @property
+    def feed(self):
+        """The :class:`FeedPanel` instance for the feed panel.
+
+        :returns: :class:`FeedPanel` instance.
+        """
+        return self._create_panel_for_id('feedPanel')
+
+    @property
+    def general(self):
+        """The :class:`GeneralPanel` instance for the general panel.
+
+        :returns: :class:`GeneralPanel` instance.
+        """
+        return self._create_panel_for_id('generalPanel')
+
+    @property
+    def media(self):
+        """The :class:`MediaPanel` instance for the media panel.
+
+        :returns: :class:`MediaPanel` instance.
+        """
+        return self._create_panel_for_id('mediaPanel')
+
+    @property
+    def panels(self):
+        """List of all the :class:`Panel` instances of the current deck.
+
+        :returns: List of :class:`Panel` instances.
+        """
+        panels = self.marionette.execute_script("""
+          let deck = arguments[0];
+          let panels = [];
+
+          for (let index = 0; index < deck.children.length; index++) {
+            panels.push(deck.children[index]);
+          }
+
+          return panels;
+        """, script_args=[self.element])
+
+        return [self._create_panel_for_id(panel.get_attribute('id')) for panel in panels]
```

Comment:
Ups, that's totally true. No need to return full DOM nodes when we can simply return the id with the `execute_script` call. I will update it.